orchestrator/test_runner/src/ica.rs

```diff
@@ -0,0 +1,816 @@
+pub fn proto_encode<M: Message>(message: &M) -> Result<Vec<u8>, EncodeError> {
+    let mut buf = Vec::with_capacity(message.encoded_len());
+    message
+        .encode(&mut buf)
+        .expect("unable to encode protobuf message");
+    Ok(buf)
+}
+
+impl AnyConvert for MsgDelegate {
+    fn from_any(value: &Any) -> Result<Self, DecodeError> {
+        assert_eq!(
+            value.type_url, STAKING_DELEGATE_TYPE_URL,
+            "invalid type url for `Any` type: expected `{}` and found `{}`",
+            STAKING_DELEGATE_TYPE_URL, value.type_url
+        );
+
+        <Self as ::prost::Message>::decode(value.value.as_slice()).map_err(Into::into)
+    }
+
+    fn to_any(&self) -> Result<Any, EncodeError> {
+        Ok(::prost_types::Any {
+            type_url: STAKING_DELEGATE_TYPE_URL.to_owned(),
+            value: proto_encode(self)?,
+        })
+    }
+}
+
+impl AnyConvert for MsgSendToEth {
+    fn from_any(value: &Any) -> Result<Self, DecodeError> {
+        assert_eq!(
+            value.type_url, MSG_SEND_TO_ETH_TYPE_URL,
+            "invalid type url for `Any` type: expected `{}` and found `{}`",
+            MSG_SEND_TO_ETH_TYPE_URL, value.type_url
+        );
+
+        <Self as ::prost::Message>::decode(value.value.as_slice()).map_err(Into::into)
+    }
+
+    fn to_any(&self) -> Result<Any, EncodeError> {
+        Ok(Any {
+            type_url: MSG_SEND_TO_ETH_TYPE_URL.to_owned(),
+            value: proto_encode(self)?,
+        })
+    }
+}
+
+/// Test Interchain accounts host / controller. Create , Send , Delegate, Send_to_Eth
+/// Plan is
+/// 1. Create and pass proposal to allow all ica_host messages <fn add_ica_host_allow_messages>
+/// 2. Get connection_id , cpc_connection_id:  <fn get_connection_id>
+/// 3. Register interchain account gravity -> ibc: <fn create_interchain_account>
+/// 4. Check account registered: <fn get_interchain_account>
+/// 5. Send some stake tokens: <fn send_tokens_to_interchain_account>, <fn get_interchain_account_balance>
+/// 6. Delegate <fn delegate_from_interchain_account>
+/// 7. Check delegations <fn check_delegatinons>
+/// 8. Adopt footoken metadata
+/// 9. SubmitTx MsgSendToEth from counterpartychain ICA. <fn submit_tx>
+/// 10. Request & submit batch
+/// 11. Check eth balance
+pub async fn ica_test(
+    contact: &Contact,
+    keys: Vec<ValidatorKeys>,
+    ibc_keys: Vec<CosmosPrivateKey>,
+    gravity_address: EthAddress,
+    web30: &Web3,
+    grpc_client: GravityQueryClient<Channel>,
+) {
+    let val_priv_keys: Vec<CosmosPrivateKey> =
+        keys.clone().into_iter().map(|k| k.validator_key).collect();
+    // create GRPC contact for counterparty chain
+    let cpc_contact = &Contact::new(&*IBC_NODE_GRPC, TIMEOUT, &*IBC_ADDRESS_PREFIX)
+        .expect("Unable to create counterparty Contact");
+
+    // Add allow messages to both chains
+    add_ica_host_allow_messages(contact, &val_priv_keys).await;
+    add_ica_host_allow_messages(cpc_contact, &ibc_keys).await;
```

Comment:
The althea-net/gaia-ica repo differs from gadost's gaia repo, so I added a governance vote to enable all messages on the ibc-chain's ICA Host module.